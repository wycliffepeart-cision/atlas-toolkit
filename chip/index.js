import { Chip } from './chip.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-chip'))
    customElements.define('atlas-chip', Chip);
