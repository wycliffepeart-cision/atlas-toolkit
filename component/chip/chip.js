import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var ChipTemplate = "<template>\n  <slot name=\"prefix\" part=\"prefix\"></slot>\n  <slot part=\"body\"></slot>\n  <slot name=\"dismisser\" part=\"dismisser\"></slot>\n  <slot name=\"suffix\" part=\"suffix\"></slot>\n</template>\n";

let Chip = class Chip extends HTMLElement {
};
Chip = __decorate([
    Template(ChipTemplate)
], Chip);
if (!customElements.get('atlas-chip'))
    customElements.define('atlas-chip', Chip);

export { Chip };
