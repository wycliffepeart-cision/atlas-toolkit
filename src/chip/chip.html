<template>
  <slot name="prefix" part="prefix"></slot>
  <slot part="body"></slot>
  <slot name="dismisser" part="dismisser"></slot>
  <slot name="suffix" part="suffix"></slot>
</template>
