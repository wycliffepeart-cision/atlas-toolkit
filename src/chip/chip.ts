import ChipTemplate from './chip.html';
import { Template } from '../decorators/template';

@Template(ChipTemplate)
export class Chip extends HTMLElement {}
