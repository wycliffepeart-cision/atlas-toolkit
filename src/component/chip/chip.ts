import ChipTemplate from './chip.html';
import { Template } from '../../decorators/template';

@Template(ChipTemplate)
export class Chip extends HTMLElement {}

if (!customElements.get('atlas-chip')) customElements.define('atlas-chip', Chip);