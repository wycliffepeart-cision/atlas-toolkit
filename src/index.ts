export * from './component/accordion/accordion';
export * from './component/alert/alert';
export * from './component/application-frame/application-frame';
export * from './component/button/button';
export * from './component/icon/icon';
export * from './component/button-set/button-set';
export * from './component/chip/chip';
export * from './component/chip-group/chip-group';
export * from './component/dialog/dialog';
export * from './component/feed/feed';
export * from './component/input-checkbox/input-checkbox';
export * from './component/input-chip-group/input-chip-group';
export * from './component/input-dropdown/input-dropdown';
export * from './component/input-search/input-search';
export * from './component/input-text/input-text';
export * from './component/input-textarea/input-textarea';
export * from './component/logo/logo';
export * from './component/main-nav/main-nav';
export * from './component/main-nav-item/main-nav-item';
export * from './component/number-display/number-display';
export * from './component/profile-control/profile-control';
export * from './component/snackbar/snackbar';
export * from './component/stepper/stepper';
export * from './component/stepper-step/stepper-step';
export * from './component/tile-board/tile-board';
export * from './component/tile-board-tile/tile-board-tile';
export * from './component/workflow/workflow';
export * from './component/workflow-step/workflow-step';
