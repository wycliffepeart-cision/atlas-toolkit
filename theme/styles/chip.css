@layer atlas {
	atlas-chip {
		--atlas-chip-base-font-size: 1em;
		--chip-body-min-width: 3ch;
		--chip-body-max-width: 100%;

		font-size: var(--atlas-chip-base-font-size);
		width: fit-content;
		height: var(--atlas-chip-body-height);
		display: flex;
		align-items: center;
		justify-content: center;

		border-radius: 9em;
		border: 1px solid var(--atlas-color-primary-n100);
		background-color: var(--atlas-color-primary-n25);
		padding-inline: 0.63em;
		padding-block: 0.38em;
	}

	atlas-chip::part(body) {
		display: inline;
		max-width: var(--chip-body-max-width);
		min-width: var(--chip-body-min-width);
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;
		text-align: start;
	}

	/* :has([slot="prefix"])::part(prefix) { */
	atlas-chip [slot="prefix"] {
		display: block;
		margin-inline-end: 1ch;
	}

	atlas-chip :is([slot="dismisser"], [slot="suffix"]) {
		font-size: calc(var(--atlas-chip-base-font-size) * 0.5);
		padding-inline-start: 0.97em;
	}

	atlas-chip[type="filter"] :is([slot="dismisser"], [slot="suffix"]) {
		font-size: calc(var(--atlas-chip-base-font-size) * 0.7);
		padding-inline-start: 0.5em;
	}

	atlas-chip atlas-button[kind="icon"] {
		color: inherit;
	}
}
