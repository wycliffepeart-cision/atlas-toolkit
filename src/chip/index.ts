import { Chip } from './chip';

if (!customElements.get('atlas-chip')) customElements.define('atlas-chip', Chip);
