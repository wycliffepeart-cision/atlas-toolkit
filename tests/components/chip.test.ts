import '../../src/chip';

const ELEMENT_TAG = 'atlas-chip';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;

afterEach(() => {
  document.body.innerHTML = '';
});

describe('Chip', () => {
  it('is correctly appended in the DOM', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    const selectedElement = document.querySelector(ELEMENT_TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toContainHTML(`<atlas-chip />`);
    (expect(element) as any).toBeEmptyDOMElement();
    (expect(getShadowDomContent(element)) as any).toMatchSnapshot();
  });

});
