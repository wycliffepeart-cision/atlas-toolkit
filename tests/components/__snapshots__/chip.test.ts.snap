// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip has the correct content when created 1`] = `
<slot name="prefix"
      part="prefix"
>
</slot>
<slot part="body">
</slot>
<slot name="dismisser"
      part="dismisser"
>
</slot>
<slot name="suffix"
      part="suffix"
>
</slot>
`;
